Doc/library/tempfile.rst

```diff
@@ -75,20 +75,36 @@ The module defines the following user-callable items:
       Added *errors* parameter.
 
 
-.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None)
-
-   This function operates exactly as :func:`TemporaryFile` does, except that
-   the file is guaranteed to have a visible name in the file system (on
-   Unix, the directory entry is not unlinked).  That name can be retrieved
-   from the :attr:`name` attribute of the returned
-   file-like object.  Whether the name can be
-   used to open the file a second time, while the named temporary file is
-   still open, varies across platforms (it can be so used on Unix; it cannot
-   on Windows).  If *delete* is true (the default), the file is
-   deleted as soon as it is closed.
+.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None, delete_on_close=True)
+
+   This function operates exactly as :func:`TemporaryFile` does, except the
+   following differences:
+
+   * The file is guaranteed to have a visible name in the file system (on Unix, the directory entry is not unlinked).
+
+   * There is more granularity in the deletion behaviour of the file
+     (see ``delete_on_close`` below)
+
    The returned object is always a file-like object whose :attr:`!file`
-   attribute is the underlying true file object. This file-like object can
-   be used in a :keyword:`with` statement, just like a normal file.
+   attribute is the underlying true file object. This file-like object can be
+   used in a :keyword:`with` statement, just like a normal file.
+
+   The name of the temporary file can be retrieved from the :attr:`name`
+   attribute of the returned file-like object. Whether the name can be used to
+   open the file a second time, while the named temporary file is still open,
+   varies across platforms (it can be so used on Unix; it cannot on Windows).
```

Comment:
In Windows, the file can be opened again if `delete_on_close` is false or if the open shares delete access (e.g. by calling `CreateFileW()` directly). If `delete_on_close` is false, and the file is opened again *without* sharing delete access (e.g. via builtin `open()`), then the second open *must* be closed before exiting the context manager, else the `unlink()` call on context manager exit will fail with a `PermissionError` due to an OS sharing-violation error (32).